nucleus/dataset.py

```diff
@@ -78,12 +95,28 @@ class Dataset:
         existing_dataset = client.get_dataset("ds_bwkezj6g5c4g05gqp1eg")
     """
 
-    def __init__(self, dataset_id, client):
+    def __init__(
+        self,
+        dataset_id: str,
+        client: "NucleusClient",  # noqa: F821
+        name: Optional[str] = None,
+    ):
+        """
+
+        Args:
+            dataset_id: Reference id of dataset
+            client: NucleusClient instance
+            name: Optionally set name on creation such that the property access doesn't need to hit the server
+        """
```

Comment:
A bit regressive, but we should remove the typehints and docstring for the `__init__()`...this is the easiest way to prevent it from showing in the API Reference. Idea is that we don't want users to think they can instantiate `Dataset` objects directly -- this should only ever happen via `NucleusClient.create_dataset`, `NucleusClient.create_dataset_from_project`, or the dashboard.

I'm sure there is a cleaner way to keep this well documented internally while hiding it from Sphinx/AutoAPI but I'm still looking :/

Reply:
That makes a lot of sense. I'll remove it and we'll see if we can find a neater way of removing it from the documentation.